src/main/java/net/minecraftforge/event/world/SleepFinishedTimeEvent.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2019.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.world;
+
+import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.eventbus.api.Cancelable;
+
+/**
+ * This event is fired when all players are asleep and the time should be set to daytime.<br>
+ *
+ * This event is cancelable. {@link Cancelable}<br>
+ *
+ * setWakeUpTime(wakeUpTime) sets a new time that will be added to the dayTime.<br>
+ */
+@Cancelable
```

Comment:
Any reason this is cancelable? The result is used even if it is canceled. 
Only difference is that if a mod cancels it, the remaining mods won't get the event. Which makes it first wins instead of last wins